engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -125,6 +126,8 @@
 	 */
 	private final List<ValidatedValueUnwrapper<?>> validatedValueHandlers;
 
+	private final ClassLoader userClassLoader;
```

Comment:
> Should we null this in close() to avoid potential class loader leaks?

sounds reasonable 

>  It can't be final then any longer though which may be a problem.

What's your concern?